web/static/js/app.js

```diff
@@ -17,22 +17,57 @@ class App {
     socket.onError( ev => console.log("ERROR", ev) )
     socket.onClose( e => console.log("CLOSE", e))
 
-    var chan = socket.channel("rooms:lobby", {})
+    var chan = socket.channel("__absinthe__:control", {})
+
+    let subscription = {
+      query: `
+      subscription Messages {
+        message(room: "lobby") {
+          body
+          author { name }
+        }
+      }
+      `,
+      variables: {}
+    }
+
     chan.join().receive("ignore", () => console.log("auth error"))
-               .receive("ok", () => console.log("join ok"))
-               .after(10000, () => console.log("Connection interruption"))
-    chan.onError(e => console.log("something went wrong", e))
-    chan.onClose(e => console.log("channel closed", e))
+      .receive("ok", () => {
+        console.log("join ok")
+
+        chan.push("doc", subscription)
+          .receive("ok", (msg) => console.log("subscription created", msg) )
+          .receive("error", (reasons) => console.log("subscription failed", reasons) )
+          .receive("timeout", () => console.log("Networking issue...") )
+      })
+      .after(10000, () => console.log("Connection interruption"))
+
 
     $input.off("keypress").on("keypress", e => {
       if (e.keyCode == 13) {
-        chan.push("new:msg", {user: $username.val(), body: $input.val()})
+        chan.push("doc", {
+          query:
+          `
+          mutation SendMessage($body: String!, $user: String!) {
+            sendMessage(room: "lobby", body: $body, user: $user) {
+              __typename
+            }
+          }
+          `,
+          variables: {
+            body: $input.val(),
+            user: $username.val()
+          }
+        })
+        .receive("ok", (msg) => console.log("mutation succeeded", msg) )
+        .receive("error", (reasons) => console.log("mutation failed", reasons) )
+        .receive("timeout", () => console.log("Networking issue...") )
         $input.val("")
       }
     })
 
-    chan.on("new:msg", msg => {
-      $messages.append(this.messageTemplate(msg))
+    chan.on("subscription:data", msg => {
```

Comment:
Hey @benwilson512 

I'm trying to adapt this example project for testing subscription-support on [apollo-phoenix-websocket](https://github.com/vic/apollo-phoenix-websocket) but I was wondering why on this line, we listen for `subscription:data` messages, looking at the absinthe subscription branch looks like its currently [hardcoded](https://github.com/absinthe-graphql/absinthe/blob/subscriptions/lib/absinthe/subscriptions.ex#L24) so I guess it's a temporary thing you had just for getting the example work. I was thinking perhaps the response from the`subscription` resolution should to the client the message-name to listen for new data coming from that subscription. 

So I suppose the subscriptions branch on absinthe is still missing something to be completed ?